workspaces/adventure-pack/goodies/typescript/Iterator.prototype.flatMap/index.test.ts

```diff
@@ -0,0 +1,184 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).flatMap;
+// eslint-disable-next-line import-x/first -- This has to happen after we delete the built-in implementation.
+import "./index";
+
+describe("Iterator.prototype.flatMap", () => {
+  it("can flatten an Array's values()", () => {
+    const array = [-1, -5, 5, 3, 7];
+    const generator = function* (p: number): Generator<number, void, void> {
+      yield p;
+      yield p * 2;
+    };
+
+    const flatMapResult = array.values().flatMap(generator);
+
+    expect([...flatMapResult]).toStrictEqual([
+      -1, -2, -5, -10, 5, 10, 3, 6, 7, 14,
+    ]);
+  });
+
+  it("can flatten a String's characters", () => {
+    const s = "hello";
+    const generator = function* (c: string): Generator<string, void, void> {
+      yield c;
+      yield c.toUpperCase();
+    };
+
+    const flatMapResult = s[Symbol.iterator]().flatMap(generator);
+
+    expect([...flatMapResult]).toStrictEqual([
+      "h",
+      "H",
+      "e",
+      "E",
+      "l",
+      "L",
+      "l",
+      "L",
+      "o",
+      "O",
+    ]);
```

Comment:
I think a small amount of programmatic expected values is ok.

You could write it as `Array.from("hHeElLlLoO")` for example to save some space.

Reply:
ack!